### ISSUE TITLE
Getting the Identity to work var xx = Container.Principal.Identity.Name;

### ISSUE THREAD

**Issue by acrigney** [NakedObjectsGroup/NakedObjectsFramework#150]
Sorry how do I get the Container.Principal.Identity.Name; to work in NO9?
The old way was to set the authentication mode in the system.web in the Web.Config
I put this in the server and the client but its not working.

<system.web>
    <authentication mode="Windows" />
    <compilation debug="true" targetFramework="4.5.2" />
    <httpRuntime targetFramework="4.5.2" />
    <roleManager enabled="true" defaultProvider="AspNetWindowsTokenRoleProvider" />
    <authorization>
      <allow users="*" />
    </authorization>
  </system.web>

**Comment by richardpawson** [comment]
Please read the NOF9 manual on Authentication.

**Comment by acrigney** [comment]
But I need this to work for windows authentication. I guess I will need to
do my own AD code?

**Comment by acrigney** [comment]
Sorry I just deployed to IIS and its not saving, I had it working before
but this time I am not getting any integrity errors in the log. Is there
another way to get errors?
I think I will create a small object and see how that goes tomorrow.
No worries we will get there.

Best Regards,
                         Alistsair

**Comment by acrigney** [comment]
Actuallyt it might be that the DTC is not setup on the UAT server, I was
told that it had been setup as now I see the old code getting an error
This is caused by the DTC not being setup.

*The partner transaction manager has disabled its support for
remote/network transactions. (Exception from HRESULT: 0x8004D025)*

**Comment by acrigney** [comment]
To use windows authentication on the NO9 client.

I put this code in the RegisterTypes function in the UnityConfig

`
container.RegisterType(typeof(IFacetDecorator), typeof(AuthorizationManager), "AuthorizationManager", new ContainerControlledLifetimeManager());
                container.RegisterInstance(typeof(IAuthorizationConfiguration),
                            NakedObjectsRunSettings.AuthorizationConfig(), new ContainerControlledLifetimeManager());

And I put this in the NakedObjectsRunSettings
public static IAuthorizationConfiguration AuthorizationConfig()
        {
            var config = new AuthorizationConfiguration<ADOAuthorizer>();
            return config;
            //return (null);
        }

And this in the NakedObjectsRunSettings

public class ADOAuthorizer : ITypeAuthorizer<object>
    {
        public IDomainObjectContainer Container { protected get; set; }
        public ADUserRepository ADUserRepository { protected get; set; }

        #region ITypeAuthorizer<object> Members

        public bool IsEditable(IPrincipal principal, object target, string memberName)
        {
            return (true);           
        }

        public bool IsVisible(IPrincipal principal, object target, string memberName)
        {
            bool isVisible = false;
            List<string> controlledCommands = new List<string>() { "SyncPractitioners", "UpdatePractitionerCourtNameCourtTypeData"};

            if (controlledCommands.Contains(memberName))
            {
                if (principal.IsInRole("LACNSW\\UWAS admin") || principal.IsInRole("LACNSW\\UWAS Power"))
                {
                    isVisible = true;
                }
            }
            else
            {
                isVisible = true;
            }
            
            return (isVisible);           
        }

        #endregion

        public void Init()
        {
            throw new NotImplementedException();
        }

        public void Shutdown()
        {
            //Does nothing
        }
    }
`

And I have my IIS server setup for windows authentication only.

But my site won't run in IIS now.  Can you help me please?

**Comment by richardpawson** [comment]
I cannot help with getting Windows authentication working on NOF9, I'm afraid -  I have no experience of this. _Much_ older versions of NOF did work OK with Windows Authentication, but that was using an all-Microsoft stack (ASP.NET Mvc).  When we went to the new SPA + REST architecture (NOF8+) we opted for an authentication approach that was very widely used with that architecture:  Auth0.  I believe that I have heard of another using switching to Windows Authentication / AD  -  but I also believe it was quite a lot of work, and don't have details.  We don't offer support on such system integration issues, sorry.
